anaconda_linter/lint/check_build_help.py

```diff
@@ -383,6 +383,24 @@ def check_recipe(self, recipe_name: str, arch_name: str, recipe: RecipeReaderDep
                 problem_paths.add(dep.path)
 
 
+class m2_must_be_updated_to_msys2(LintCheck):
+    """
+    The m2 tool {} should be updated to msys2.
```

Comment:
`m2-*` are just a set of packages with a common prefix.  It/they are not a tool.

```suggestion
    The m2-* packages {} should be updated to msys2-*.
```

Reply:
Thank you, fixed!